cdap-app-fabric/src/main/java/co/cask/cdap/reporting/ProgramHeartbeatDataset.java

```diff
@@ -1,193 +0,0 @@
- *
- * http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
- * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
- * License for the specific language governing permissions and limitations under
- * the License.
- */
-
-package co.cask.cdap.reporting;
-
-import co.cask.cdap.api.common.Bytes;
-import co.cask.cdap.api.data.DatasetContext;
-import co.cask.cdap.api.dataset.DatasetManagementException;
-import co.cask.cdap.api.dataset.lib.AbstractDataset;
-import co.cask.cdap.api.dataset.table.Row;
-import co.cask.cdap.api.dataset.table.Scanner;
-import co.cask.cdap.api.dataset.table.Table;
-import co.cask.cdap.api.dataset.table.TableProperties;
-import co.cask.cdap.common.conf.CConfiguration;
-import co.cask.cdap.common.conf.Constants;
-import co.cask.cdap.data2.datafabric.dataset.DatasetsUtil;
-import co.cask.cdap.data2.dataset2.DatasetFramework;
-import co.cask.cdap.data2.dataset2.lib.table.MDSKey;
-import co.cask.cdap.internal.app.runtime.schedule.TriggeringScheduleInfoAdapter;
-import co.cask.cdap.internal.app.store.RunRecordMeta;
-import co.cask.cdap.proto.ProgramType;
-import co.cask.cdap.proto.id.DatasetId;
-import co.cask.cdap.proto.id.NamespaceId;
-import co.cask.cdap.proto.id.ProgramRunId;
-import com.google.gson.Gson;
-import com.google.gson.GsonBuilder;
-
-import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
-
-/**
- * Heartbeat Store that writes heart beat messages and program status messages
- * to program heartbeat table. This is used for efficiently
- * scanning and returning results for dashboard status queries
- *
- * Rowkey for the table is organized in the following format
- * [namespace][timestamp-in-seconds][program-run-Id]
- *
- */
-public class ProgramHeartbeatDataset extends AbstractDataset {
-  private static final Gson GSON = TriggeringScheduleInfoAdapter.addTypeAdapters(new GsonBuilder()).create();
-  private static final byte[] COLUMN_NAME = Bytes.toBytes("status");
-  private static final DatasetId PROGRAM_HEARTBEAT_INSTANCE_ID =
-    NamespaceId.SYSTEM.dataset(Constants.ProgramHeartbeat.TABLE);
-  private final Table table;
-
-  public ProgramHeartbeatDataset(Table table) {
-    super("ignored", table);
-    this.table = table;
-  }
-
-  /**
-   * Static method for creating an instance of {@link ProgramHeartbeatDataset}.
-   */
-  public static ProgramHeartbeatDataset getOrCreate(DatasetContext datasetContext,
-                                                    DatasetFramework datasetFramework, CConfiguration cConfiguration) {
-    try {
-      TableProperties.Builder props = TableProperties.builder();
-      long ttl = cConfiguration.getLong(Constants.ProgramHeartbeat.HEARTBEAT_TABLE_TTL_SECONDS);
-      props.setTTL(ttl);
```

Comment:
Is there a way to set ttl in new spis? if not, how the clean up will happen?

Reply:
hmm that's a good point - I didn't see this. It won't - so if needed we need to be able to do this in the SPI somehow

Reply:
discussed with albert, added todo

Reply:
sg